File: MenaCart/Domain/Models/Category.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Models
{
    public class Category
    {
        [Key]
        public int CategoryId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        public int? ParentCategoryId { get; set; }

        // Navigation
        [ForeignKey(nameof(ParentCategoryId))]
        public Category ParentCategory { get; set; }

        public ICollection<Category> ChildCategories { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}

File: menacart-frontend/src/types/category.ts
export interface Category {
  categoryId: number;
  name: string;
  parentCategoryId: number | null;
  imageUrl?: string | null;
  parentCategoryName: string | null;
  childCategories: Category[];
}

export interface CreateCategoryRequest {
  name: string;
  imageUrl?: string | null;
  parentCategoryId?: number;
}

export interface UpdateCategoryRequest {
  name?: string;
  imageUrl?: string | null;
  parentCategoryId?: number;
}
